### ISSUE TITLE
dnd-moved not firing

### ISSUE THREAD

**Issue by jmls** [marceljuenemann/angular-drag-and-drop-lists#153]
I have an element defined as this in the html

```html
<li layout-fill ng-repeat="item in items" 
    dnd-draggable="item" 
    dnd-type="item'" 
    dnd-effect-allowed="move" 
    dnd-moved="console.log('fooooo')">
```

however, the dnd-moved callback never gets called. The "move" works because the dragged item is inserted

what am I missing ?

**Comment by marceljuenemann** [comment]
Did you define console in your scope?

**Comment by dantheman999301** [comment]
I'm having a similar issue, specifically when I'm using dnd-moved and dnd-drop together.

I have two nested lists, each with dnd-moved and dnd-drop on them. For some reason on the second list, dnd-moved doesn't fire, but it does on the first one.

e.g.

```html            
<ul class="col-xs-12 category-list"
    dnd-list="shortlist.Categories"
    dnd-allowed-types="['category']"
    dnd-drop="dropCategory(event, index, item, external, type)">
         <li class="category-container col-xs-12"
         ng-repeat="shortlistCat in shortlist.Categories"
         dnd-draggable="shortlistCat"
         dnd-type="shortlistCat.type"
         dnd-moved="moveCategory($index)"
         dnd-effect-allowed="move">
             <div class="category-header" ng-class="getShortlistColourClass(shortlistCat.Colour)">
                 <div class="grey-bottom">
                     {{shortlistCat.Name}}
                 </div>
             </div>
             <ul class="col-xs-12 player-list"
                 dnd-list="shortlistCat.Players"
                 dnd-allowed-types="['player']"
                 dnd-drop="dropPlayer(event, index, item, external, type, shortlistCat)">
                 <li ng-repeat="player in shortlistCat.Players"
                     dnd-draggable="player"
                     dnd-type="player.type"
                     dnd-moved="console.log('Derp')"
                     dnd-effect-allowed="move">
```

**Comment by dantheman999301** [comment]
Writing this out as a second comment as the behaviour is really odd.

Apparently they can both work at the same time nest but if you put a breakpoint on either of the drop functions, dnd-move is never called. If you remove the breakpoint, everything goes back to normal. So I'm guessing it's an odd timing issue.

**Comment by nibblesnbits** [comment]
Has this issue ever been actually resolved?  I've having exactly the same problem.

**Comment by dantheman999301** [comment]
My problem was that I combined this with dnd-dropped and put a break point
in dnd-dropped.

For some reason when adding a breakpoint there, it caused moved not to be
fired. The only way I could figure out to solve it was to either set some
scope variables and then use them in the moved function instead or wrap the
code in dropped in a setTimeout to debug it. This would cause it to happen
after dnd-moved but everything seemed to work OK.

I did step through the library code and it seemed to be the browser that
wasn't firing correctly rather than the library from what I remember.

Would be interested to see if anyone can solve this properly.
